### PR TITLE
Make ForeighHash inherit the default value of the referenced Hash

### PR DESCRIPTION
**Pull request by Endogen**
## Description

Make ForeighHash inherit the default value of the referenced Hash

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have tested this change in my development environment.
- [ ] I have added tests to prove that this change works
- [ ] All existing tests pass after this change
- [ ] I have added / updated documentation related to this change

### CODE REVIEW

**Comment by Endogen** [comment]
Needs to be tested

**Comment by Endogen** [comment]
@ duelingbenjos I've tested it. Works fine.

File: src/contracting/storage/orm.py [hunk 1]
```diff
@@ -38,6 +38,10 @@ def __init__(self, contract, name, driver: Driver = driver, default_value=None):
         self._delimiter = constants.DELIMITER
         self._default_value = default_value
 
+        # Store the default_value in storage if it's not None
+        if default_value is not None:
+            self._driver.set(f'{self._key}{self._delimiter}__default__', default_value)
```
Comment (duelingbenjos) [inline src/contracting/storage/orm.py:43]: should `__default__` be in a constants file or similar somewhere ?

**Review by duelingbenjos — CHANGES_REQUESTED** [review]

File: src/contracting/storage/orm.py (hunk 1 above)
Comment (Endogen) [reply src/contracting/storage/orm.py:43]: Yea probably a good idea. I will take another look at it

**Review by crosschainer — APPROVED** [review]

**Review by duelingbenjos — APPROVED** [review]

**Pull request merged**